File: project-root/asm-msa-transaction/src/main/java/com/company/asm/transaction/service/TransactionService.java
package com.company.asm.transaction.service;

import com.company.asm.transaction.api.model.TransactionInput;
import com.company.asm.transaction.api.model.TransactionOutput;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public interface TransactionService {
    TransactionOutput createTransaction(TransactionInput transactionInput);
    TransactionOutput getTransactionById(UUID transactionId);
    List<TransactionOutput> getTransactionsByFilter(UUID accountId, LocalDate startDate, LocalDate endDate);
    List<TransactionOutput> getAllTransactions();
}

File: project-root/app-spa-portal/src/app/features/transaction/services/transaction/transaction.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ITransaction } from '../../models/transaction.model';
import { SERVICE_URL_ENDPOINTS } from '../../../../core/config/service-url-endpoints';
import { ITransactionParams } from '../../models/transaction-params.model';

@Injectable({
  providedIn: 'root'
})
export class TransactionService {
  constructor(private readonly http: HttpClient) { }

  getTransactionById(idTransaction: string) {
    return this.http.get<ITransaction>(`${SERVICE_URL_ENDPOINTS.TRANSACTION}/${idTransaction}`);
  }

  getTransactionByAccountIdAndDate(params: ITransactionParams) {
    const httpParams = new HttpParams()
      .set('accountId', params.accountId)
      .set('startDate', params.startDate)
      .set('endDate', params.endDate);
    return this.http.get<ITransaction[]>(SERVICE_URL_ENDPOINTS.TRANSACTIONS_FILTER, { params: httpParams });
  }

  createTransaction(data: ITransaction) {
    return this.http.post<ITransaction>(SERVICE_URL_ENDPOINTS.TRANSACTION_CREATE, data);
  }

  getTransactions() {
    return this.http.get<ITransaction[]>(SERVICE_URL_ENDPOINTS.TRANSACTIONS_ALL);
  }

}
